framework/util/page_guard_manager.cpp

```diff
@@ -498,43 +518,83 @@ void PageGuardManager::AddExceptionHandler()
             exception_handler_count_ = 1;
         }
 #else
-        // Retrieve the current SIGSEGV handler info before replacing the current signal handler to determine if our
-        // replacement signal handler should use an alternate signal stack.
-        int result = sigaction(MEMPROT_SIGNAL, nullptr, &s_old_sigaction);
-
-        if (result != -1)
+#if defined(__ANDROID__)
+        if (!libsigchain_active_ && AddSpecialSignalHandlerFn)
         {
-            struct sigaction sa = {};
-
-            sa.sa_flags = SA_SIGINFO;
-            sigemptyset(&sa.sa_mask);
-            sa.sa_sigaction = PageGuardExceptionHandler;
+            static std::jmp_buf abort_env;
```

Comment:
It is referenced in the `sa_sigaction` callback so it needs a static lifetime. I could have also made it a global variable, but as you say, it only gets used in this function. If there is a better way to pass `abort_env` to the `SIGABRT` handler I would love to know!

Reply:
ah, I see it is to make the lambda assigned to sa_sigaction to be stateless, so it can be assigned to a function pointer.
Thanks for the explanation :-)
No, I'm not aware of a better way to pass an value to a signal handler.